module/mempool/stdmap/backdata/arrayBackData.go

```diff
@@ -0,0 +1,400 @@
+package backdata
+
+import (
+	"encoding/binary"
+	"time"
+	_ "unsafe" // for linking runtimeNano
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/mempool/stdmap/backdata/heropool"
+)
+
+//go:linkname runtimeNano runtime.nanotime
+func runtimeNano() int64
+
+const (
+	bucketSize = uint64(16)
+
+	// telemetryCounterInterval is the number of required interactions with
+	// this back data prior to printing any log. This is done as a slow-down mechanism
+	// to avoid spamming logs upon read/write heavy operations. An interaction can be
+	// a read or write.
+	telemetryCounterInterval = uint64(1000)
+
+	// telemetryDurationInterval is the required elapsed duration interval
+	// prior to printing any log. This is done as a slow-down mechanism
+	// to avoid spamming logs upon read/write heavy operations.
+	telemetryDurationInterval = 10 * time.Second
+)
+
+// bIndex is data type representing a bucket index.
+type bIndex uint64
+
+// sIndex is data type representing a slot index in a bucket.
+type sIndex uint64
+
+// idPrefix is a 32-bits prefix flow.Identifier used to determine the bIndex of the entity
+// it represents.
+type idPrefix uint32
+
+type key struct {
```

Comment:
Would it be better to call this a `slot` rather than a `key`? Most of the other comments already refer to bucket slots... And the struct is more of a key reference and does not contain the key itself? So if not `slot` then maybe `keyRef` and update all the existing slot comments?

Reply:
https://github.com/onflow/flow-go/pull/1778/commits/4c6272dc08dfc1fcd252b9e03c6348e965b947e6